tests/sources/rest_api/integration/test_processing_steps.py

```diff
@@ -259,3 +260,141 @@ def extend_body(row):
 
     data = list(mock_source.with_resources("comments"))
     assert data[0]["body"] == "Post 2 - Comment 0 for post 2"
+
+
+def flatten_reactions(post):
+    post_without_reactions = copy.deepcopy(post)
+    post_without_reactions.pop("reactions")
+    for reaction in post["reactions"]:
+        yield {"reaction": reaction, **post_without_reactions}
+
+
+def test_rest_api_source_yield_map(mock_api_server) -> None:
+    config: RESTAPIConfig = {
+        "client": {
+            "base_url": "https://api.example.com",
+        },
+        "resources": [
+            {
+                "name": "posts",
+                "endpoint": "posts_with_reactions",
+                "processing_steps": [
+                    {"yield_map": flatten_reactions},
+                ],
+            },
+        ],
+    }
+    mock_source = rest_api_source(config)
+
+    data = list(mock_source.with_resources("posts"))
+
+    assert all("reaction" in record and "reactions" not in record for record in data)
```

Comment:
this may be nitpicking but it makes sense to assert expected records count in all tests. here I think it is:
DEFAULT_PAGE_SIZE * DEFAULT_TOTAL_PAGES*DEFAULT_REACTIONS_COUNT right?

same for other tests. such assert is makes sure that records were correctly flattened and all expected data got extracted